std/algorithm/searching.d

```diff
@@ -1210,6 +1210,8 @@ bool endsWith(alias pred, R)(R doesThisEnd)
     }
 }
 
+private struct IdentityFun {};
```

Comment:
Please add a comment here describing why this exists. Also, the common D style is to put helpers after the function, as this style was necessitated by C compilers and is unnecessary in the modern age.